### ISSUE TITLE
ID selectors starting with valid HEX will get highlighted.

### ISSUE THREAD

**Issue by CasperBraske** [Monnoroch/ColorHighlighter#396]
Using words like `#back` and `#featured` make the beginning of those words get highlighted.

I'm using Ubuntu 16.04 and Sublime build 3143.

**Comment by Monnoroch** [comment]
Yep, this is a bug. Would you perhaps suggest better regexes to handle it?

**Comment by CasperBraske** [comment]
Well, I don't know. It's quite hard to define.
I actually thought they were applied by scope. For the HEX values, they are:

source.css meta.property-list.css meta.property-value.css constant.other.color.rgb-value.css

**Comment by Manifold0** [comment]
Adding a quick check for a word boundary or a semicolon does the trick locally. I'm still running through some more test cases.
`(\b|;)`

`"#(?P<R>hex)(?P<G>hex)(?P<B>hex)(\b|;)"`

**Comment by Monnoroch** [comment]
`;` should also be `\b`, so why `|;`?

**Comment by Manifold0** [comment]
Testing locally using just the `\b` on the end of the regex doesn't yield a match. Adding in the `|;` gets it running. Not sure if they python regex parsing is not catching the `\b` in the right place or what.

Here is with just `\b`
<img width="212" alt="screen shot 2017-10-16 at 10 43 22 am" src="https://user-images.githubusercontent.com/2730575/31618065-ee2c7750-b25e-11e7-8624-04763579999e.png">

**Comment by Manifold0** [comment]
`\b` should work, according to my understanding of Python regex

https://regex101.com/r/uzapkV/2

**Comment by Monnoroch** [comment]
@Manifold0 the problem with `;` is that it's tailored to `#AABBCC;`. What if I write `#AABBCC    ;`? Or what if it's not CSS?

**Comment by Manifold0** [comment]
`#ABC ;` catches on the `\b` part of `(\b|;)` and ignores the `;`

The order matters, if you have `(;|\b)` then you'll get the `;` back as a match, but having it as the second alternative means that catching the `\b` will return the capture group.

I can try to debug the issue with `\b` not matching on its own (that would be the best solution that I can think of, because hex can be treating as a word for this case), but I don't quite know where to begin with debugging a Sublime package.

**Comment by Monnoroch** [comment]
@Manifold0 my problem with `;` is that it's too CSS-specific. The plugin is mainly targeted at stylesheets but also works on any other kind of file. So having `;` in the regex is just a hack.

**Comment by Manifold0** [comment]
I'm totally on board with that. I'll keep tinkering with it and come back if I find something more workable and less specific.
